### ISSUE TITLE
Move ELRS protocol code to separate lib

### ISSUE THREAD

**Issue by wvarty** [AlessandroAU/ExpressLRS#65]
The ELRS packet encoding/decoding currently lives in the tx_main.cpp and rx_main.cpp files. This has a number of downsides, including:

* Common code is duplicated across TX and RX, has to be modified in 2 places
* Very difficult / impossible to unit test
* Protocol code is hard to find amongst the TX/RX code
* Protocol functions are only accessible from within tx_main.cpp or rx_main.cpp, which leads to a bad habit where all functional code is written in those files, leading to them becoming unmanageable.

This issue is a proposal to move the ELRS protocol code out to a separate library.
The lib would likely be designed to accept CRSF packets, and output ELRS packets, to be sent to the LoRa Radio class.

**Comment by JyeSmith** [comment]
Makes sense and even more so when we end up with multiple switch options.  This should lead to rx and tx_main being easier to read.
